kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsCloud.java

```diff
@@ -34,7 +38,23 @@ public String getAwsCmd() {
 
     @Override
     public boolean isAvailable() {
-        return !Environment.AWS_ACCESS_KEY_ID.equals(Environment.AWS_ACCESS_KEY_ID_DEFAULT);
+        boolean awsCloudSelected = Environment.AWS_USE_CLOUD.equalsIgnoreCase("true");
+        boolean keyIdDefaulted = Environment.AWS_ACCESS_KEY_ID.equals(Environment.AWS_ACCESS_KEY_ID_DEFAULT);
+        boolean kroxyliciousKeyIdDefaulted = Environment.AWS_KROXYLICIOUS_ACCESS_KEY_ID.equals(Environment.AWS_KROXYLICIOUS_ACCESS_KEY_ID_DEFAULT);
+        if (awsCloudSelected && !keyIdDefaulted && !kroxyliciousKeyIdDefaulted) {
```

Comment:
Do we actually get any benefit from the default checking and warnings?

AFAIK LocalStack doesn't care what access id/secret you use, so why not just whatever the user has specified in the environment through to it.     That'll give us a single code path for AWS and LocalStack.

This comment is the best documentation I could fine.  I also tested against the localstack container.

https://github.com/localstack/localstack/issues/62#issuecomment-294749459

Reply:
> Do we actually get any benefit from the default checking and warnings?

From my POV, doing that we inform if some missconfiguration has happened and run by default LocalStack. We could also just check `AWS_USE_CLOUD` and run the test accordingly. In that case wrong keys will make the test fail because we cannot create keys and so on. Maybe that's a best option to mark the user error when configuring the test parameters.

> AFAIK LocalStack doesn't care what access id/secret you use, so why not just whatever the user has specified in the environment through to it. That'll give us a single code path for AWS and LocalStack.

I don't know if it is directly related or not. The LocalStack documentation talks about with kind of access keys are rejected: https://docs.localstack.cloud/references/configuration/#:~:text=PARITY_AWS_ACCESS_KEY_ID

I believe something has changed since that comment (it is from 2017).

Reply:
> From my POV, doing that we inform if some missconfiguration has happened and run by default LocalStack. 

If they have opted for Cloud and don't provide valid access keys, the failure will be clear enough.  Does adding more validation actually buy us much?

I don't **think** [PARITY_AWS_ACCESS_KEY_ID](https://docs.localstack.cloud/references/configuration/#:~:text=PARITY_AWS_ACCESS_KEY_ID) is relevant to us.  We're not using LocalStack to create access keys.

> I believe something has changed since that comment (it is from 2017). Anyway I will test it

Thank you.  I experimented on the command line.  LocalStack didn't seem to care whatever accessid/secretkey I provided.  

```
kwall@Oslo ~ % AWS_ACCESS_KEY_ID=foo AWS_SECRET_KEY=bar aws --endpoint-url http://localhost:4566  sts get-caller-identity
{
    "UserId": "AKIAIOSFODNN7EXAMPLE",
    "Account": "000000000000",
    "Arn": "arn:aws:iam::000000000000:root"
}
```

Reply:
>That's worse. That'll mean that a misconfiguration goes unnoticed. We'll think tests are running against the Cloud when they aren't.

yeah I agree. We should just let it run and fail rather than revert to localstack.